aggregator/src/test/java/org/jenkinsci/plugins/workflow/BuildQueueTasksTest.java

```diff
@@ -0,0 +1,192 @@
+import hudson.model.queue.QueueTaskFuture;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+public class BuildQueueTasksTest {
+
+    @Rule public RestartableJenkinsRule story = JenkinsRuleExt.workAroundJenkins30395Restartable();
+
+    @Issue("JENKINS-28649")
+    @Test public void queueAPI() throws Exception {
+        // This is implicitly testing ExecutorStepExecution$PlaceholderTask as exported bean
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                // use non-existent node label to keep the build queued
+                p.setDefinition(new CpsFlowDefinition("node('linux') { echo 'test' }"));
+
+                WorkflowRun b = scheduleAndWaitQueued(p);
+                assertQueueAPIStatusOKAndAbort(b);
+            }
+        });
+    }
+
+    @Issue("JENKINS-28649")
+    @Test public void queueAPIRestartable() throws Exception {
+        // This is implicitly testing AfterRestartTask as exported bean
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                // use non-existent node label to keep the build queued
+                p.setDefinition(new CpsFlowDefinition("node('linux') { echo 'test' }"));
+                scheduleAndWaitQueued(p);
+                // Ok, the item is in he queue now, restart
+            }
+        });
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
+                WorkflowRun b = p.getBuildByNumber(1);
+
+                assertQueueAPIStatusOKAndAbort(b);
+            }
+        });
+    }
+
+    @Issue("JENKINS-28649")
+    @Test public void computerAPI() throws Exception {
+        // This is implicitly testing ExecutorStepExecution$PlaceholderTask$PlaceholderExecutable as exported bean
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                        "  echo 'test'\n " +
+                        "  semaphore 'watch'\n " +
+                        "}"));
+
+                WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+                SemaphoreStep.waitForStart("watch/1", b);
+
+                assertComputerAPIStatusOK();
+
+                SemaphoreStep.success("watch/1", null);
+                story.j.waitUntilNoActivity();
+            }
+        });
+    }
+
+    @Issue("JENKINS-28649")
+    @Test public void computerAPIRestartable() throws Exception {
```

Comment:
:ant: why not just fold this into the previous test? `waitForStart`, do your assertions, restart, do your assertions again.